osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -119,9 +123,12 @@ private void load(OsuConfigManager config)
                 return;
             }
 
+            if (DrawableRuleset is IDrawableScrollingRuleset scrollingRuleset)
+                dependencies.CacheAs(scrollingRuleset.ScrollingInfo);
```

Comment:
This is another example: if you have to use soft casts to subclasses in the base class, then that's a strong sign that maybe the logic should live in the subclass to begin with.

Reply:
Or that we shouldn't have subclasses in the first place and should use this pattern more often, I dunno. Hierarchy has been pretty restrictive in these kinds of usages.

Reply:
I'm gonna look away from this one for the time being because I kind of want to try and switch the scrolling stuff to composition too at some point. And it's not even precedent (see `FindSnappedPositionAndTime()`).